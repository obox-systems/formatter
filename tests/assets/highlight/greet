macro_rules! greet {
    () => {
        println!("Hello, world!");
    };
    ($name:expr) => {
        println!("Hello, {}!", $name);
    };
}

macro_rules! calculate_sum {
    ($($x:expr),*) => {
        {
            let mut sum = 0;
            $(
                sum += $x;
            )*
            sum
        }
    };
}

fn main() {
    greet!(); // Prints "Hello, world!"
    greet!("Alice"); // Prints "Hello, Alice!"

    let num1 = 5;
    let num2 = 10;
    let num3 = 15;

    let sum = calculate_sum!(num1, num2, num3);
    println!("The sum is: {}", sum); // Prints "The sum is: 30"
}
