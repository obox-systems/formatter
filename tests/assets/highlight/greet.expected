[42mmacro_rules[0m[40m![0m[47m [0m[42mgreet[0m[47m [0m[100m{[0m[101m
[0m[47m    [0m[100m([0m[100m)[0m[47m [0m[42m=>[0m[47m [0m[100m{[0m[101m
[0m[47m        [0m[42mprintln[0m[40m![0m[100m([0m[105m"Hello, world!"[0m[100m)[0m[40m;[0m[101m
[0m[47m    [0m[100m}[0m[40m;[0m[101m
[0m[47m    [0m[100m([0m[40m$[0m[42mname[0m[42m:[0m[42mexpr[0m[100m)[0m[47m [0m[42m=>[0m[47m [0m[100m{[0m[101m
[0m[47m        [0m[42mprintln[0m[40m![0m[100m([0m[105m"Hello, {}!"[0m[40m,[0m[47m [0m[40m$[0m[42mname[0m[100m)[0m[40m;[0m[101m
[0m[47m    [0m[100m}[0m[40m;[0m[101m
[0m[100m}[0m[101m

[0m[42mmacro_rules[0m[40m![0m[47m [0m[42mcalculate_sum[0m[47m [0m[100m{[0m[101m
[0m[47m    [0m[100m([0m[40m$[0m[100m([0m[40m$[0m[42mx[0m[42m:[0m[42mexpr[0m[100m)[0m[40m,[0m[42m*[0m[100m)[0m[47m [0m[42m=>[0m[47m [0m[100m{[0m[101m
[0m[47m        [0m[100m{[0m[101m
[0m[47m            [0m[42mlet[0m[47m [0m[42mmut[0m[47m [0m[42msum[0m[47m [0m[42m=[0m[47m [0m[40m0[0m[40m;[0m[101m
[0m[47m            [0m[40m$[0m[100m([0m[101m
[0m[47m                [0m[42msum[0m[47m [0m[42m+=[0m[47m [0m[40m$[0m[42mx[0m[40m;[0m[101m
[0m[47m            [0m[100m)[0m[42m*[0m[101m
[0m[47m            [0m[42msum[0m[101m
[0m[47m        [0m[100m}[0m[101m
[0m[47m    [0m[100m}[0m[40m;[0m[101m
[0m[100m}[0m[101m

[0m[42mfn[0m[47m [0m[42mmain[0m[100m([0m[100m)[0m[47m [0m[100m{[0m[101m
[0m[47m    [0m[42mgreet[0m[40m![0m[100m([0m[100m)[0m[40m;[0m[47m [0m[46m// Prints "Hello, world!"[0m[101m
[0m[47m    [0m[42mgreet[0m[40m![0m[100m([0m[105m"Alice"[0m[100m)[0m[40m;[0m[47m [0m[46m// Prints "Hello, Alice!"[0m[101m

[0m[47m    [0m[42mlet[0m[47m [0m[42mnum1[0m[47m [0m[42m=[0m[47m [0m[40m5[0m[40m;[0m[101m
[0m[47m    [0m[42mlet[0m[47m [0m[42mnum2[0m[47m [0m[42m=[0m[47m [0m[40m1[0m[40m0[0m[40m;[0m[101m
[0m[47m    [0m[42mlet[0m[47m [0m[42mnum3[0m[47m [0m[42m=[0m[47m [0m[40m1[0m[40m5[0m[40m;[0m[101m

[0m[47m    [0m[42mlet[0m[47m [0m[42msum[0m[47m [0m[42m=[0m[47m [0m[42mcalculate_sum[0m[40m![0m[100m([0m[42mnum1[0m[40m,[0m[47m [0m[42mnum2[0m[40m,[0m[47m [0m[42mnum3[0m[100m)[0m[40m;[0m[101m
[0m[47m    [0m[42mprintln[0m[40m![0m[100m([0m[105m"The sum is: {}"[0m[40m,[0m[47m [0m[42msum[0m[100m)[0m[40m;[0m[47m [0m[46m// Prints "The sum is: 30"[0m[101m
[0m[100m}[0m[101m
[0m